shadows/framework/src/main/java/org/robolectric/shadows/ImageUtil.java

```diff
@@ -46,6 +51,37 @@ public static Point getImageSizeFromStream(InputStream is) {
     }
   }
 
+  public static RobolectricBufferedImage getImageFromStream(InputStream is) {
+    if (!initialized) {
+      // Stops ImageIO from creating temp files when reading images
+      // from input stream.
+      ImageIO.setUseCache(false);
+      initialized = true;
+    }
+
+    try {
+      ImageInputStream imageStream = createImageInputStream(is);
+      Iterator<ImageReader> readers = ImageIO.getImageReaders(imageStream);
+      if (!readers.hasNext()) return null;
+
+      ImageReader reader = readers.next();
+      try {
+        reader.setInput(imageStream);
+        String mimeType = reader.getFormatName();
+        int minIndex = reader.getMinIndex();
+        BufferedImage image = reader.read(minIndex);
+        RobolectricBufferedImage robolectricBufferedImage = new RobolectricBufferedImage();
+        robolectricBufferedImage.bufferedImage = image;
+        robolectricBufferedImage.mimeType = mimeType;
```

Comment:
Can you replace the mime logic with something like this:

```
        String format = reader.getFormatName();
        int minIndex = reader.getMinIndex();
        BufferedImage image = reader.read(minIndex);
        RobolectricBufferedImage robolectricBufferedImage = new RobolectricBufferedImage();
        robolectricBufferedImage.bufferedImage = image;
        robolectricBufferedImage.mimeType = "image/"+format;
        return robolectricBufferedImage;
```

Determining mime type from extension requires reading a system file, and some JVM's may not have access to that file. Since there are only 2-3 image types we can just manually do it.

Reply:
I use `("image/" + format).toLowerCase()`, because the format is the upper case on my machine. Use `toLowerCase()` directly can ensure the final format is the format used by other test codes.